mlibrary/search
# Add `utm_source` query parameter to external links.

# Overview
For analytics tracking of external links, the query parameter `utm_source=library-search` needed to be added. The `Anchor` component has been created. If the component has the `to` attribute, it will be treated as a `Link`. If the component has the `href` attribute, it will be treated as a traditional anchor. If the `href` is defined, it will check:
1. Does it link to a website (versus `mailto`, `tel`, etc.)?
2. Does the link not have the same domain as the current site?
If the above two are true, then the query parameter will be added to the URL. From there, Google Analytics will be able to track the activity.

All files that had a `Link` component or an `a` tag have been replaced with the `Anchor` component. Files that returned strings with anchors inside have been rewritten to return HTML instead.

The only remaining external links are the `Report a problem` links, coming from Spectrum, but [the fix has been submitted](https://github.com/mlibrary/spectrum/pull/157).

## Anything else?
### Class to Funcational Components
In an ongoing effort to convert the class components into functional components to follow today's [React practices](https://react.dev/learn/your-first-component), these components have all been rewritten as they are in relation to the use of the `Anchor` component:
* `BrowseAtoZ `
* `NestedList`
* `BrowseFilter`
* `Node`
* `BrowseByFilters`
* `BrowseInfo`
* `GoToList`
* `BentoboxList`
* `KeywordSwitch`
* `Record`
* `FullRecordBreadcrumbs`
* `RecordPreview`

## Testing
- Make sure the PR is consistent in these browsers:
- Run accessibility tests:
- Go through the site.
  - Is anything broken?
  - Do the external links have the `utm_source=library-search` query parameter?
  - Are there any external links that do not have the parameter?
  - Do all internal links not have the parameter?
